pkg/controllers/scheduler/schedulingtriggers.go

```diff
@@ -303,6 +303,7 @@ var knownSchedulingAnnotations = sets.New(
 	AffinityAnnotations,
 	MaxClustersAnnotations,
 	FollowsObjectAnnotation,
+	common.NoSyncAnnotation,
```

Comment:
Why do we reschedule after disabling/enabling dryrun?

Reply:
This feels unexpected. After disabling dryrun, the user may simply want to sync the current scheduling result.